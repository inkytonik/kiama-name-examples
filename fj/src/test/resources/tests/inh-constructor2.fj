class A extends Object {
  X x;
  A(X x) {
    super();
    this.x = x;
  }
}
// Require that the parameter type is the same as
// the field's type in the supertype
class B extends A {
  B(Y x) {
    super(x);
  }
}
class X extends Object {
  X() {
    super();
  }
}
class Y extends X {
  Y() {
    super();
  }
}
