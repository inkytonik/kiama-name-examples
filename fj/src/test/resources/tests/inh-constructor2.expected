src/test/resources/tests/inh-constructor2.fj:2:5:info: x def: X
  X x;
    ^
src/test/resources/tests/inh-constructor2.fj:3:7:info: x def: X
  A(X x) {
      ^
src/test/resources/tests/inh-constructor2.fj:11:3:error: constructor args (Y x) should be the fields (X x)
  B(Y x) {
  ^
src/test/resources/tests/inh-constructor2.fj:11:7:info: x def: Y
  B(Y x) {
      ^
src/test/resources/tests/inh-constructor2.fj:12:11:error: super arg 0 type must be X, not Y
    super(x);
          ^
